src/internal/kopia/wrapper.go

```diff
@@ -710,3 +713,184 @@ func (w Wrapper) setMaintenanceParams(
 
 	return nil
 }
+
+// SetRetentionParameters sets configuration values related to immutable backups
+// and retention policies on the storage bucket. The minimum retention period
+// must be >= 24hrs due to kopia default expectations about full maintenance.
+func (w *Wrapper) SetRetentionParameters(
+	ctx context.Context,
+	retention repository.Retention,
+) error {
+	if retention.Mode == nil && retention.Duration == nil && retention.Extend == nil {
+		return nil
+	}
+
+	// Somewhat confusing case, when we have no retention but a non-zero duration
+	// it acts like we passed in only the duration and returns an error about
+	// having to set both. Return a clearer error here instead.
+	if ptr.Val(retention.Mode) == repository.NoRetention && ptr.Val(retention.Duration) != 0 {
+		return clues.New("retention duration must be 0 or unset if disabling retention")
+	}
+
+	dr, ok := w.c.Repository.(repo.DirectRepository)
+	if !ok {
+		return clues.New("unable to get valid handle to repo").WithClues(ctx)
```

Comment:
Per error standards, please avoid phrases like "unable to foo"
```suggestion
		return clues.New("getting handle to repo").WithClues(ctx)
```